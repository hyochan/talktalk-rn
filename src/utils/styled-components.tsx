import * as styledComponents from "styled-components/native";
import ThemeInterface from "./theme";

const {
  default: styled,
  css,
  ThemeProvider
} = styledComponents as styledComponents.ReactNativeThemedStyledComponentsModule<typeof ThemeInterface>;

export { css, ThemeProvider };
export default styled;