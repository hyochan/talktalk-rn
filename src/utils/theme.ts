import { Dimensions, Platform } from 'react-native';
import { getStatusBarHeight } from 'react-native-status-bar-height';
import { DefaultTheme } from 'styled-components';

const { width, height } = Dimensions.get('window');
let calRatio = width <= height ? 16 * (width / height) : 16 * (height / width);
if (width <= height) {
  if (calRatio < 9) {
    calRatio = width / 9;
  } else {
    calRatio = height / 18;
  }
} else {
  if (calRatio < 9) {
    calRatio = height / 9;
  } else {
    calRatio = width / 18;
  }
}

const theme: DefaultTheme = {
  screenWidth: width,
  screenHeight: height,
  ratio: calRatio / (360 / 9),
  statusBarHeight: getStatusBarHeight(true),
  colors: {
    background: '#e3e3e3',
    dodgerBlue: 'rgb(58,139,255)',
    dusk: 'rgb(65,77,107)',
    blueyGray: 'rgb(134,154,183)',
    cloudyBlue: 'rgb(175,194,219)',
    greenishCyan: 'rgb(80,227,194)',
    paleGray: 'rgb(233,237,244)',
  }
};

export default theme;