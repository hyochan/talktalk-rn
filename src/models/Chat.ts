import { User } from './User';

export class Chat {
  public constructor(
    public id: string = '', 
    public sender: User = undefined, 
    public message: string = '', 
    public created: Date = undefined, 
    public updated: Date = undefined
  ) {}
}
