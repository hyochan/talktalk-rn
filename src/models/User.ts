import { ImageURISource } from 'react-native';
import { Chatroom } from './Chatroom';

export class User {
  public constructor(
    public uid: string = '',
    public displayName: string = '',
    public photoURL: string = '',
    public statusMsg: string = '',
    public isOnline: boolean = false,
    public friends: User[] = [],
    public chatrooms: Chatroom[] = [],
    public created: Date = undefined,
    public updated: Date = undefined,
  ) {}
}
