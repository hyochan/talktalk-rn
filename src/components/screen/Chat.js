// @flow
import React, {
  Component,
  useState,
  useEffect,
  useRef,
} from 'react';
import {
  SafeAreaView,
  StyleSheet,
  TouchableOpacity,
  Image,
  Text,
  View,
  Keyboard,
  Platform,
  TextInput,
  KeyboardAvoidingView,
  FlatList,
} from 'react-native';
import styled from 'styled-components/native';

import { Header } from 'react-navigation';

import type {
  ____ViewStyleProp_Internal as ViewStyle,
  ____TextStyleProp_Internal as TextStyle,
  ____ImageStyleProp_Internal as ImageStyle,
} from 'react-native/Libraries/StyleSheet/StyleSheetTypes';

import ChatListItem from '../shared/ChatListItem';
import EmptyListItem from '../shared/EmptyListItem';
import Button from '../shared/Button';

import { Chatroom } from '../../models/Chatroom';
import { IC_SMILE } from '../../utils/Icons';
import { ratio, colors } from '../../utils/Styles';
import { getString } from '../../../STRINGS';
import { Chat } from '../../models/Chat';
import { User } from '../../models/User';

const StyledContainer = styled.SafeAreaView`
  flex: 1;
  background-color: white;
  flex-direction: column;
  align-items: center;
`;

const StyledKeyboardAvoidingView = styled.KeyboardAvoidingView`
  flex: 1;
  justify-content: center;
  align-self: stretch;

  flex-direction: column;
  align-items: center;
`;

const StyledViewChat = styled.View`
  width: 100%;
  border-top-width: 0.5;
  border-color: rgb(225,225,225);
  background-color: white;
  min-height: 52;
  max-height: 100;

  flex-direction: row;
  justify-content: flex-end;
  align-items: center;
`;

const StyledInputChat = styled.TextInput`
  width: 80%;
  font-size: 14;
  margin-right: 20;
  padding-left: 48;
  color: black;
`;

const StyledTouchMenu = styled.TouchableOpacity`
  position: absolute;
  left: 10;
  height: 100%;
  minWidth: 20;
  justifyContent: center;
`;

const StyledViewBottom = styled.View`
  position: absolute;
  bottom: 0;
  width: 100%;
`;

const StyledViewMenu = styled.View`
  height: 258;
  background-color: green;
`;

type Styles = {
  btnSend: ViewStyle,
  txtSend: TextStyle,
};

const styles: Styles = StyleSheet.create({
  btnSend: {
    right: 8,
    backgroundColor: colors.dodgerBlue,
    borderRadius: 4,
    width: 60,
    height: 36,

    alignItems: 'center',
    justifyContent: 'center',
  },
  txtSend: {
    fontSize: 14,
    fontWeight: 'bold',
    color: 'white',
    paddingHorizontal: 5,
    paddingVertical: 10,
  },
});

type Props = {
  navigation: any;
};
type State = {
  isLoading: boolean,
  showMenu: boolean,
  message: string,
  chats: Chat[],
};

function Screen(props: Props, state: State) {
  let keyboardDidShowListener: any;
  let keyboardDidHideListener: any;

  const input1 = useRef(null);
  const input2 = useRef(null);
  const [isLoading, setIsLoading] = useState(false);
  const [showMenu, setShowMenu] = useState(false);
  const [message, setMessage] = useState('');
  const [chats, setChats] = useState([
    new Chat(
      '',
      new User(
        '0',
        'sender111',
        '',
      ),
      'hello1',
    ),
    new Chat(
      '',
      new User(
        '0',
        'sender111',
        '',
      ),
      'Hello2. This is long message. This is long message. This is long message. This is long message. This is long message. This is long message. This is long message. This is long message. This is long message. This is long message. This is long message.',
    ),
    new Chat(
      '',
      new User(
        '1',
        'sender2222',
        '',
      ),
      'hello',
    ),
    new Chat(
      '',
      new User(
        '2',
        'sender2222',
        '',
      ),
      'hello',
    ),
    new Chat(
      '',
      new User(
        '2',
        'sender2222',
        '',
      ),
      'hello. This is my long message. This is my long message. This is my long message. This is my long message. This is my long message. This is my long message. This is my long message. This is my long message. This is my long message.',
    ),
  ]);

  useEffect(() => {
    if (showMenu) {
      Keyboard.dismiss();
    } else {
      if (input1 && input1.current) {
        input1.current.focus();
      }
    }
  }, [showMenu]);

  useEffect(() => {
    Keyboard.addListener('keyboardDidShow', (e) => {
      console.log('keyboardHeight', e.endCoordinates.height);
    });
    return () => {
      if (keyboardDidShowListener) {
        keyboardDidShowListener.remove();
      }
    };
  });

  const renderItem = ({ item, index } : any) => {
    return (
      <ChatListItem
        prevItem={index > 0 ? chats[ index - 1 ] : null}
        item={item}
      />
    );
  };

  const sendChat = () => {
    console.log('sendChat');
  };

  const goBack = () => {
    props.navigation.goBack();
  };

  return (
    <StyledContainer>
      <StyledKeyboardAvoidingView
        keyboardVerticalOffset = {Header.HEIGHT + 24}
        behavior={Platform.select({
          'android': null,
          'ios': 'padding',
        })}
      >
        <FlatList
          style={{ alignSelf: 'stretch' }}
          contentContainerStyle={
            chats.length === 0
              ? {
                flex: 1,
                alignItems: 'center',
                justifyContent: 'center',
              }
              : null
          }
          keyExtractor={(item, index) => index.toString()}
          data={chats}
          renderItem={renderItem}
          ListEmptyComponent={<EmptyListItem>{getString('NO_CONTENT')}</EmptyListItem>}
        />
        {
          !showMenu
            ? <StyledViewChat>
              <StyledInputChat
                ref={input1}
                onFocus={() => setShowMenu(false)}
                multiline={true}
                placeholder={ getString('WRITE_MESSAGE') }
                placeholderTextColor={ colors.cloudyBlue }
                value={message}
                defaultValue={message}
                onChangeText={(text) => setMessage(text)}
              />
              <StyledTouchMenu
                onPress={() => setShowMenu(true)}
              >
                <Image style={{
                  width: 20,
                  height: 20,
                }} source={IC_SMILE}/>
              </StyledTouchMenu>
              <Button
                isLoading={isLoading}
                onPress={sendChat}
                style={styles.btnSend}
                textStyle={styles.txtSend}
              >{getString('SEND')}</Button>
            </StyledViewChat>
            : null
        }
      </StyledKeyboardAvoidingView>
      {
        showMenu
          ? <StyledViewBottom>
            <StyledViewChat>
              <StyledInputChat
                ref={input2}
                onFocus={() => setShowMenu(false)}
                multiline={true}
                placeholder={ getString('WRITE_MESSAGE') }
                placeholderTextColor={ colors.cloudyBlue }
                value={message}
                defaultValue={message}
              />
              <StyledTouchMenu
                onPress={() => setShowMenu(false)}
              >
                <Image style={{
                  width: 20,
                  height: 20,
                }} source={IC_SMILE}/>
              </StyledTouchMenu>
              <Button
                isLoading={isLoading}
                onPress={sendChat}
                style={styles.btnSend}
                textStyle={styles.txtSend}
              >{getString('SEND')}</Button>
            </StyledViewChat>
            <StyledViewMenu />
          </StyledViewBottom>
          : null
      }
    </StyledContainer>
  );
}

export default Screen;
