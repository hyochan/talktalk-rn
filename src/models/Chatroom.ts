import { Chat } from './Chat';
import { User } from './User';

export class Chatroom {
  public constructor(
    public id: string = '',
    public lastChat: Chat = undefined,
    public lastChatCnt: number = 0,
    public chats: Chat[] = [],
    public users: User[] = [],
    public created: Date = undefined,
    public updated: Date = undefined,
  ) {}
}
