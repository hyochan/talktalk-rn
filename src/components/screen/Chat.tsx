import React, {
  Component,
  useState,
  useEffect,
  useRef,
} from 'react';
import {
  SafeAreaView,
  StyleSheet,
  TouchableOpacity,
  Image,
  Text,
  View,
  Keyboard,
  Platform,
  TextInput,
  KeyboardAvoidingView,
  FlatList,
} from 'react-native';
import { NavigationScreenProp, NavigationStateRoute } from 'react-navigation';
import styled, { ThemeProps, withTheme, DefaultTheme } from 'styled-components/native';

import { Header } from 'react-navigation';
import Icon5, { FA5Style } from 'react-native-vector-icons/FontAwesome5';
import ChatListItem from '../shared/ChatListItem';
import EmptyListItem from '../shared/EmptyListItem';
import Button from '../shared/Button';

import { Chatroom } from '../../models/Chatroom';
import { IC_SMILE } from '../../utils/Icons';
import { getString } from '../../../STRINGS';
import { Chat } from '../../models/Chat';
import { User } from '../../models/User';

const StyledContainer = styled.SafeAreaView`
  flex: 1;
  background-color: white;
  flex-direction: column;
  align-items: center;
`;

const StyledKeyboardAvoidingView = styled.KeyboardAvoidingView`
  flex: 1;
  justify-content: center;
  align-self: stretch;
  flex-direction: column;
  align-items: center;
`;

const StyledViewChat = styled.View`
  width: 100%;
  border-top-width: 0.5;
  border-color: rgb(225,225,225);
  background-color: white;
  min-height: 52;
  max-height: 100;
  flex-direction: row;
  justify-content: flex-end;
  align-items: center;
`;

const StyledInputChat = styled.TextInput`
  width: 80%;
  font-size: 14;
  margin-right: 20;
  padding-left: 48;
  color: black;
`;

const StyledTouchMenu = styled.TouchableOpacity`
  position: absolute;
  left: 10;
  height: 100%;
  min-width: 20px;
  justify-content: center;
`;

const StyledViewBottom = styled.View`
  position: absolute;
  bottom: 0;
  width: 100%;
`;

const StyledViewMenu = styled.View<{ height: number }>`
  background-color: ${({ theme: { colors: { paleGray }}}) => paleGray};
  flex-direction: row;
  flex-wrap: wrap;
  height: ${({ height }) => height};
`;

interface IProps extends ThemeProps<DefaultTheme> {
  navigation: any;
}

interface IState {
  isLoading: boolean;
  showMenu: boolean;
  message: string;
  chats: Chat[];
}

function Screen(props: IProps, state: IState) {
  let keyboardShowListener: any;

  const input1 = useRef(null);
  const input2 = useRef(null);
  const [keyboardHeight, setKeyboardHeight] = useState(258);
  const [isLoading, setIsLoading] = useState(false);
  const [showMenu, setShowMenu] = useState(false);
  const [message, setMessage] = useState('');
  const [chats, setChats] = useState([
    new Chat(
      '',
      new User(
        '0',
        'sender111',
        '',
      ),
      'hello1',
    ),
    new Chat(
      '',
      new User(
        '0',
        'sender111',
        '',
      ),
      'Hello2. This is long message. This is long message. This is long message.This is long message. This is long message. This is long message. This is long message. This is long message. This is long message. This is long message. This is long message.',
    ),
    new Chat(
      '',
      new User(
        '1',
        'sender2222',
        '',
      ),
      'hello',
    ),
    new Chat(
      '',
      new User(
        '2',
        'sender2222',
        '',
      ),
      'hello',
    ),
    new Chat(
      '',
      new User(
        '2',
        'sender2222',
        '',
      ),
      'hello. This is my long message. This is my long message. This is my long message. This is my long message. This is my long message. This is my long message. This is my long message. This is my long message. This is my long message.',
    ),
  ]);

  useEffect(() => {
    if (showMenu) {
      Keyboard.dismiss();
    } else {
      if (input1 && input1.current) {
        input1.current.focus();
      }
    }
  }, [showMenu]);

  useEffect(() => {
    keyboardShowListener = Keyboard.addListener('keyboardDidShow', (e) => {
      setKeyboardHeight(e.endCoordinates.height);
    });
    return () => {
      if (keyboardShowListener) {
        keyboardShowListener.remove();
      }
    };
  });

  const renderItem = ({ item, index }: any) => {
    return (
      <ChatListItem
        prevItem={index > 0 ? chats[ index - 1 ] : null}
        item={item}
      />
    );
  };

  const sendChat = () => {
  };

  const {
    theme: {
      colors: {
        dusk,
        cloudyBlue
      }
    }    
  } = props;

  return (
    <StyledContainer>
      <StyledKeyboardAvoidingView
        keyboardVerticalOffset={Header.HEIGHT}
        behavior={Platform.select({
          android: null,
          ios: 'padding',
        })}
      >
        <FlatList
          style={{ alignSelf: 'stretch' }}
          contentContainerStyle={
            chats.length === 0
              ? {
                flex: 1,
                alignItems: 'center',
                justifyContent: 'center',
              }
              : null
          }
          keyExtractor={(item, index) => index.toString()}
          data={chats}
          renderItem={renderItem}
          ListEmptyComponent={<EmptyListItem>{getString('NO_CONTENT')}</EmptyListItem>}
        />
        {
          !showMenu
            ? <StyledViewChat>
              <StyledInputChat
                testID='input_chat'
                ref={input1}
                onFocus={() => setShowMenu(false)}
                multiline={true}
                placeholder={ getString('WRITE_MESSAGE') }
                placeholderTextColor={cloudyBlue}
                value={message}
                defaultValue={message}
                onChangeText={(text) => setMessage(text)}
              />
              <StyledTouchMenu
                testID='touch_menu'
                onPress={() => setShowMenu(true)}
              >
                <Image style={{
                  width: 20,
                  height: 20,
                }} source={IC_SMILE}/>
              </StyledTouchMenu>
              <Button
                testID='btn_chat'
                isLoading={isLoading}
                onPress={sendChat}
              >{getString('SEND')}</Button>
            </StyledViewChat>
            : null
        }
      </StyledKeyboardAvoidingView>
      {
        showMenu
          ? <StyledViewBottom>
            <StyledViewChat>
              <StyledInputChat
                ref={input2}
                onFocus={() => setShowMenu(false)}
                multiline={true}
                placeholder={ getString('WRITE_MESSAGE') }
                placeholderTextColor={cloudyBlue}
                value={message}
                defaultValue={message}
              />
              <StyledTouchMenu
                testID='touch_menu'
                onPress={() => setShowMenu(false)}
              >
                <Image style={{
                  width: 20,
                  height: 20,
                }} source={IC_SMILE}/>
              </StyledTouchMenu>
              <Button
                testID='btn_chat'
                isLoading={isLoading}
                onPress={sendChat}
              >{getString('SEND')}</Button>
            </StyledViewChat>
            <StyledViewMenu testID="viewMenu" height={keyboardHeight}>
              <View style={{
                flexDirection: 'row',
                marginTop: 10,
              }}>
                <TouchableOpacity
                  style={{
                    marginLeft: 16,
                    marginTop: 2,
                    width: 60,
                    height: 60,
                    justifyContent: 'center',
                    alignItems: 'center',
                  }}
                >
                  <Icon5 name='camera' size={36} color={dusk} light />
                </TouchableOpacity>
                <TouchableOpacity
                  style={{
                    marginLeft: 16,
                    marginTop: 4,
                    width: 60,
                    height: 60,
                    justifyContent: 'center',
                    alignItems: 'center',
                  }}
                >
                  <Icon5 name='image' size={40} color={dusk} light/>
                </TouchableOpacity>
              </View>
            </StyledViewMenu>
          </StyledViewBottom>
          : null
      }
    </StyledContainer>
  );
}

export default withTheme(Screen);
