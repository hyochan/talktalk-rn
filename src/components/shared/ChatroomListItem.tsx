import React, { Component } from 'react';
import {
  StyleSheet,
  Image,
  Text,
  View,
  TouchableOpacity,
  ViewStyle,
  TextStyle,
} from 'react-native';

import styled from 'styled-components/native';

import moment from 'moment';
import Icon5 from 'react-native-vector-icons/FontAwesome5';

import { Chatroom } from '../../models/Chatroom';
import { colors } from '../../utils/Styles';
import { Chat } from '../../models/Chat';
import { User } from '../../models/User';
import { BaseButton } from 'react-native-gesture-handler';

interface IStyles {
  wrapper: ViewStyle;
  txtMsg: TextStyle;
  txtDate: TextStyle;
}

const styles: IStyles = StyleSheet.create({
  wrapper: {
    backgroundColor: 'white',
    height: 80,
    borderBottomWidth: 1,
    borderColor: 'rgb(247,248,251)',

    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'flex-start',
  },
  txtMsg: {
    fontSize: 12,
    color: colors.dusk,
    maxWidth: 150,
  },
  txtDate: {
    fontSize: 12,
    color: colors.blueyGray,
    textAlign: 'right',
  },
});

const StyledWrapper = styled.View`
  background-color: white;
  height: 80;
  border-bottom-width: 1;
  border-color: rgb(247,248,251);

  flexDirection: row;
  alignItems: center;
  justifyContent: flex-start;
`;

const StyledStatus = styled.View`
  position: absolute;
  width: 12;
  height: 12;
  borderRadius: 6;
  backgroundColor: ${colors.greenishCyan};
  right: 0;
  bottom: 0;
  border-width: 2;
  border-color: white;
`;

const StyledViewContent = styled.View`
  flex-direction: column;
  flex: 1;
  padding-right: 20;
`;

const StyledViewTop = styled.View`
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
`;

const StyledTextDisplayName = styled.Text`
  font-weight: bold;
  font-size: 14;
  color: ${colors.dusk};
`;

const StyledViewCount = styled.View`
  width: 16;
  height: 16;
  border-radius: 8;
  justify-content: center;
  align-items: center;
  background-color: rgb(80, 227, 194);
`;

const StyledTextCount = styled.Text`
  font-size: 10;
  color: white;
`;

const StyledViewBottom = styled.View`
  margin-top: 8;

  flex-direction: row;
  align-items: center;
  justify-content: space-between;
`;

const StyledTextMessage = styled.Text`
  font-size: 12;
  color: ${colors.dusk};
  max-width: 150;
`;

const StyledTextDate = styled.Text`
  font-size: 12;
  color: ${colors.blueyGray};
  text-align: right;
`;

interface IProps {
  testID?: string;
  style?: ViewStyle;
  item: Chatroom;
  onPress?: () => void;
}

function Shared(props: IProps) {
  return (
    <View style={{ width: '100%' }}>
      <TouchableOpacity
        activeOpacity={0.5}
        onPress={props.onPress}
      >
        <View style={props.style}>
          <View style={{
            marginHorizontal: 20,
          }}>
            {
              props.item.lastChat.sender.photoURL
                ? <Image style={{
                  width: 40,
                  height: 40,
                }} source={{ uri: props.item.lastChat.sender.photoURL }}/>
                : <Icon5 name='meh' size={40} color={colors.dusk} light/>
            }
            {
              props.item.lastChat.sender.isOnline
                ? <StyledStatus/>
                : <View/>
            }
          </View>
          <StyledViewContent>
            <StyledViewTop>
              <StyledTextDisplayName>{props.item.lastChat.sender.displayName}</StyledTextDisplayName>
              {
                props.item.lastChatCnt !== 0
                  ? <StyledViewContent>
                    <StyledTextCount>{props.item.lastChatCnt}</StyledTextCount>
                  </StyledViewContent>
                  : <View/>
              }
            </StyledViewTop>
            <StyledViewBottom>
              <Text
                numberOfLines={2}
                style={[
                  styles.txtMsg,
                  props.item.lastChatCnt !== 0 // Have unread message status
                    ? { fontWeight: 'bold' }
                    : { },
                ]}
              >{props.item.lastChat.message}</Text>
              <Text style={styles.txtDate}>{
                props.item.lastChat.created
                  ? moment(props.item.lastChat.created).fromNow()
                  : 'nan'
              }</Text>
            </StyledViewBottom>
          </StyledViewContent>
        </View>
      </TouchableOpacity>
    </View>
  );
}

Shared.defaultProps = {
  style: styles.wrapper,
  item: new Chatroom(
    '',
    new Chat(
      '',
      new User('', '', '', ''),
      '',
    ),
  ),
};

export default Shared;
