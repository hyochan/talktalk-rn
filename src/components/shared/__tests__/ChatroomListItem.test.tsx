import 'react-native';
import * as React from 'react';
import ChatroomListItem, { styles } from '../ChatroomListItem';
import { ThemeProvider } from 'styled-components/native';
import theme from '../../../utils/theme';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { render, fireEvent } from 'react-native-testing-library';
import { Chatroom } from '../../../models/Chatroom';
import { Chat } from '../../../models/Chat';
import { User } from '../../../models/User';

const props = {
  style: styles.wrapper,
  item: new Chatroom(
    '',
    new Chat(
      '',
      new User('', '', '', ''),
      '',
    ),
  )
}

const component: React.ReactElement = (
  <ThemeProvider theme={theme}>
    <ChatroomListItem {...props} />
  </ThemeProvider>
);

describe('[ChatroomListItem] rendering test', () => {
  it('renders as expected', () => {
    const json = renderer.create(component).toJSON();
    expect(json).toMatchSnapshot();
  });
});
