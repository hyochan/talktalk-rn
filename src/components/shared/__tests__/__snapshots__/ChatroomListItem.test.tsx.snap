// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ChatroomListItem] rendering test renders as expected 1`] = `
<View
  style={
    Object {
      "width": "100%",
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "borderBottomWidth": 1,
          "borderColor": "rgb(247,248,251)",
          "flexDirection": "row",
          "height": 80,
          "justifyContent": "flex-start",
        }
      }
    >
      <View
        style={
          Object {
            "marginHorizontal": 20,
          }
        }
      >
        <Unknown
          color="rgb(65,77,107)"
          light={true}
          name="meh"
          size={40}
        />
        <View />
      </View>
      <View
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexDirection": "column",
              "flexGrow": 1,
              "flexShrink": 1,
              "paddingRight": 20,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "rgb(65,77,107)",
                  "fontSize": 14,
                  "fontWeight": "bold",
                },
              ]
            }
          >
            
          </Text>
          <View
            style={
              Array [
                Object {
                  "flexBasis": 0,
                  "flexDirection": "column",
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "paddingRight": 20,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 10,
                  },
                ]
              }
            />
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
                "marginTop": 8,
              },
            ]
          }
        >
          <Text
            numberOfLines={2}
            style={
              Array [
                Object {
                  "color": "rgb(65,77,107)",
                  "fontSize": 12,
                  "maxWidth": 150,
                },
                Object {
                  "fontWeight": "bold",
                },
              ]
            }
          >
            
          </Text>
          <Text
            style={
              Object {
                "color": "rgb(134,154,183)",
                "fontSize": 12,
                "textAlign": "right",
              }
            }
          >
            nan
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
